src/util/schnorr.rs

```diff
@@ -18,64 +18,114 @@
 //!
 
 pub use secp256k1::schnorrsig::{PublicKey, KeyPair};
-use secp256k1::{Secp256k1, Verification};
+use secp256k1::{Secp256k1, Verification, Error, constants, Signing};
 use hashes::{Hash, HashEngine};
 use util::taproot::{TapBranchHash, TapTweakHash};
+use core::{fmt, str};
 
 /// Untweaked Schnorr public key
 pub type UntweakedPublicKey = PublicKey;
 
 /// Tweaked Schnorr public key
+#[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug)]
 pub struct TweakedPublicKey(PublicKey);
 
+impl fmt::LowerHex for TweakedPublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::LowerHex::fmt(&self.0, f)
+    }
+}
+
+impl fmt::Display for TweakedPublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl str::FromStr for TweakedPublicKey {
```

Comment:
Isn't this kinda dangerous? If one accidentally parses untweaked key it'd be a problem. Maybe we should have distinct encoding for tweaked keys? (e.g. prefix them with `!tweaked!:`)

Reply:
Yes. I think that introducing new prefix/encoding will be controversial and will delay reviewing of the PR, so I just simply removed this - and let's have a separate discussion. I personally prefer to use some form of bech32m encoding with custom prefix